### PR TITLE
Refactor: added type annotation

### PR DESCRIPTION
**Pull request by AliMuhammadOfficial**
Refactor: added type annotation
---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

**Comment by potiuk** [comment]
Some static chceks are failling. I highly recommend to install pre-commit locally to fix them automatically

File: tests/build_provider_packages_dependencies.py [hunk 1]
```diff
@@ -114,13 +114,14 @@ class ImportFinder(NodeVisitor):
     AST visitor that collects all imported names in its imports
     """
 
-    def __init__(self, filename):
+    def __init__(self, filename: str) -> None:
+        
         self.imports: List[str] = []
-        self.filename = filename
+        self.filename: str = filename
```
Comment (uranusjr) [inline tests/build_provider_packages_dependencies.py]: I don’t think this is needed, Mypy is able to infer this from the argument type IIRC.

File: tests/build_provider_packages_dependencies.py [hunk 2]
```diff
@@ -114,13 +114,14 @@ class ImportFinder(NodeVisitor):
     AST visitor that collects all imported names in its imports
     """
 
-    def __init__(self, filename):
+    def __init__(self, filename: str) -> None:
+
```
Comment (uranusjr) [inline tests/build_provider_packages_dependencies.py]:
```suggestion
    def __init__(self, filename: str) -> None:
```

(suggested by flake8)

**Review by uranusjr — CHANGES_REQUESTED** [review]

**Review by uranusjr — APPROVED** [review]
Pylint is still complaining but that’s not for this PR to handle 😞

**Pull request merged**